DatadogTrace/Tests/OpenTelemetry/OTelSpanTests.swift

```diff
@@ -0,0 +1,220 @@
+
+        // When
+        childSpan.end()
+        parentSpan.end()
+
+        // Then
+        waitForExpectations(timeout: 0.5, handler: nil)
+        let events: [SpanEventsEnvelope] = core.events()
+        XCTAssertEqual(events.count, 2)
+        let child = events.first!.spans.first!
+        let parent = events.last!.spans.first!
+        XCTAssertEqual(parent.parentID, nil)
+        XCTAssertEqual(child.parentID, parent.spanID)
+    }
+
+    func testSetNoParent() {
+        let writeSpanExpectation = expectation(description: "write span event")
+        writeSpanExpectation.expectedFulfillmentCount = 2
+        let core = PassthroughCoreMock(expectation: writeSpanExpectation)
+
+        // Given
+        let tracer: DatadogTracer = .mockWith(core: core)
+        let parentSpan = tracer.spanBuilder(spanName: "Parent").startSpan()
+        _ = tracer.spanBuilder(spanName: "Noise").startSpan()
+        let childSpan = tracer.spanBuilder(spanName: "Child").setNoParent().startSpan()
+
+        // When
+        childSpan.end()
+        parentSpan.end()
+
+        // Then
+        waitForExpectations(timeout: 0.5, handler: nil)
+        let events: [SpanEventsEnvelope] = core.events()
+        XCTAssertEqual(events.count, 2)
+        let child = events.first!.spans.first!
+        let parent = events.last!.spans.first!
+        XCTAssertEqual(parent.parentID, nil)
+        XCTAssertEqual(child.parentID, nil)
+    }
+
+    func testSetAttribute() {
+        let writeSpanExpectation = expectation(description: "write span event")
+        let core = PassthroughCoreMock(expectation: writeSpanExpectation)
+
+        // Given
+        let tracer: DatadogTracer = .mockWith(core: core)
+        let span = tracer.spanBuilder(spanName: "Span").startSpan()
+
+        // When
+        span.setAttribute(key: "key", value: .bool(true))
+        span.setAttribute(key: "key2", value: .string("value2"))
+        span.setAttribute(key: "key3", value: .int(3))
+        span.setAttribute(key: "key4", value: .double(4.0))
+
+        span.end()
+
+        // Then
+        waitForExpectations(timeout: 0.5, handler: nil)
+        let events: [SpanEventsEnvelope] = core.events()
+        XCTAssertEqual(events.count, 1)
+        let recordedSpan = events.first!.spans.first!
+        let expectedTags =
+        [
+            "key": "true",
+            "key2": "value2",
+            "key3": "3",
+            "key4": "4.0",
+            "span.kind": "client",
+        ]
+        XCTAssertTagsEqual(recordedSpan.tags, expectedTags)
+    }
+}
+
+func XCTAssertTagsEqual(
+    _ dict1: [String: String],
+    _ dict2: [String: String],
+    file: StaticString = #filePath,
+    line: UInt = #line
+) {
+    XCTAssertEqual(dict1.count, dict2.count, file: file, line: line)
+    for (key, value) in dict1 {
+        XCTAssertEqual(
+            dict2[key],
+            value,
+            "Expected \(key) to be \(value), but was \(String(describing: dict2[key]))",
```

Comment:
nice addition! I hate reading assertion details on bigger objects.

Maybe worth putting to `TestUtilities`

Reply:
suggestion/ Such assert is already part of `TestUtilities`:

- [`public func DDAssertDictionariesEqual(_:_:_:file:line:)`](https://github.com/DataDog/dd-sdk-ios/blob/c91484f5c71039b8ee464e9c697707c0634f4dfe/TestUtilities/Helpers/DDAssert.swift#L194)

let's use it instead.